lib/gcpspanner/spanneradapters/backend.go

```diff
@@ -560,6 +545,79 @@ func (s *Backend) GetSavedSearch(ctx context.Context, savedSearchID string, user
 		return nil, err
 	}
 
+	return convertUserSavedSearchToSavedSearchResponse(savedSearch), nil
+}
+
+func buildUpdateSavedSearchRequestForGCP(savedSearchID string,
+	userID string,
+	updateRequest *backend.SavedSearchUpdateRequest) gcpspanner.UpdateSavedSearchRequest {
+	req := gcpspanner.UpdateSavedSearchRequest{
+		ID:       savedSearchID,
+		AuthorID: userID,
+		Query: gcpspanner.OptionallySet[string]{
+			IsSet: false,
+			Value: "",
+		},
+		Name: gcpspanner.OptionallySet[string]{
+			IsSet: false,
+			Value: "",
+		},
+		Description: gcpspanner.OptionallySet[*string]{
+			IsSet: false,
+			Value: nil,
+		},
+	}
+	if slices.Contains(updateRequest.UpdateMask, backend.SavedSearchUpdateRequestMaskName) {
+		req.Name.IsSet = true
+		req.Name.Value = *updateRequest.Name
```

Comment:
Should we check for empty strings? Or the question is how can we tell if a user is trying to clear a field vs just leaving it empty on the UI? e.g. a user intends to only update `query` but left `name` empty. UI probably needs to distinguish between clearing a field vs leaving it empty. 

But either case, the name probably shouldn't be empty?

Reply:
actually never mind, I saw that there is a `minLength: 1` on the `name` field. The UI question stands for the `description` field but unrelated to this PR!

Reply:
Good question! I'm actually going to put the minLength back to 1 on description. and then add [nullable](https://swagger.io/docs/specification/v3_0/data-models/data-types/#null) to it. This will make it clearer on which fields can truly be cleared